src/Authenticator/CookieAuthenticator.php

```
@@ -0,0 +1,121 @@
+ */
+class CookieAuthenticator extends AbstractAuthenticator implements PersistenceInterface
+{
+
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(IdentifierCollection $identifiers, array $config = [])
+    {
+        $this->_checkCakeVersion();
+
+        $this->_defaultConfig['cookie'] = [
+            'name' => 'CookieAuth',
+            'expire' => null,
+            'path' => '/',
+            'domain' => '',
+            'secure' => false,
+            'httpOnly' => false,
+            'value' => ''
+        ];
+
+        parent::__construct($identifiers, $config);
+    }
+
+    /**
+     * Checks the CakePHP Version by looking for the cookie implementation
+     *
+     * @return void
+     */
+    protected function _checkCakeVersion()
+    {
+        if (!class_exists('Cake\Http\Cookie\Cookie')) {
+            throw new RuntimeException('You must use the CakePHP `3.next` branch or wait until CakePHP version 3.5 is released to use the CookieAuthenticator');
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function authenticate(ServerRequestInterface $request, ResponseInterface $response)
+    {
+        $userData = $request->getCookie($this->getConfig('cookie.name'));
+
+        if (empty($userData)) {
+            return new Result(null, Result::FAILURE_CREDENTIALS_NOT_FOUND, [
+                'Login credentials not found'
+            ]);
+        }
+
+        $user = $this->identifiers()->identify($userData);
+
+        if (empty($user)) {
+            return new Result(null, Result::FAILURE_IDENTITY_NOT_FOUND, $this->identifiers()->getErrors());
+        }
+
+        return new Result($user, Result::SUCCESS);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function persistIdentity(ServerRequestInterface $request, ResponseInterface $response, $identity)
+    {
+        $cookie = $this->getConfig('cookie');
+        $cookie['value'] = $identity;
```

I would just throw an exception.